Uncommitted files cause error in other Sphinx extensions, because context["sourcename"] is removed

Thanks for this neat extension to Sphinx.

I use Sphinx with the [Sphinx Book Theme](https://sphinx-book-theme.readthedocs.io/) and the *sphinx-last-updated-by-git* extension. When I create a new file in my documentation project that is not yet committed to the Git repository and I run `make html`, I see the following traceback:

```
[app] emitting event: 'html-page-context'('usage', 'page.html', {'embedded': False, 'project': '…
 
[app] emitting event: 'build-finished'(ExtensionError("Handler <function add_header_buttons at 0x7ff53dcfd4c0> for event 'html-page-contex
sitemap.xml was generated for URL / in /home/…/docs/_build/html/sitemap.xml

Traceback (most recent call last):
  File "/home/…/.virtualenvs/documentation/lib/python3.9/site-packages/sphinx/events.py", line 94, in emit
    results.append(listener.handler(self.app, *args))
  File "/home/…/.virtualenvs/documentation/lib/python3.9/site-packages/sphinx_book_theme/header_buttons/__init__.py", line 152, in add_header_buttons
    "url": f'{pathto("_sources", 1)}/{context["sourcename"]}',
KeyError: 'sourcename'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/…/.virtualenvs/documentation/lib/python3.9/site-packages/sphinx/cmd/build.py", line 276, in build_main
    app.build(args.force_all, filenames)
  File "/home/…/.virtualenvs/documentation/lib/python3.9/site-packages/sphinx/application.py", line 330, in build
    self.builder.build_update()
  File "/home/…/.virtualenvs/documentation/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 286, in build_update
    self.build(to_build,
  File "/home/…/.virtualenvs/documentation/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 350, in build
    self.write(docnames, list(updated_docnames), method)
  File "/home/…/.virtualenvs/documentation/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 524, in write
    self._write_serial(sorted(docnames))
  File "/home/…/.virtualenvs/documentation/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 534, in _write_serial
    self.write_doc(docname, doctree)
  File "/home/…/.virtualenvs/documentation/lib/python3.9/site-packages/sphinx/builders/html/__init__.py", line 625, in write_doc
    self.handle_page(docname, ctx, event_arg=doctree)
  File "/home/…/.virtualenvs/documentation/lib/python3.9/site-packages/sphinx/builders/html/__init__.py", line 1026, in handle_page
    newtmpl = self.app.emit_firstresult('html-page-context', pagename,
  File "/home/…/.virtualenvs/documentation/lib/python3.9/site-packages/sphinx/application.py", line 457, in emit_firstresult
    return self.events.emit_firstresult(event, *args,
  File "/home/…/.virtualenvs/documentation/lib/python3.9/site-packages/sphinx/events.py", line 112, in emit_firstresult
    for result in self.emit(name, *args, allowed_exceptions=allowed_exceptions):
  File "/home/…/.virtualenvs/documentation/lib/python3.9/site-packages/sphinx/events.py", line 102, in emit
    raise ExtensionError(__("Handler %r for event %r threw an exception") %
sphinx.errors.ExtensionError: Handler <function add_header_buttons at 0x7ff53dcfd4c0> for event 'html-page-context' threw an exception (exception: 'sourcename')

Extension error (sphinx_book_theme.header_buttons):
Handler <function add_header_buttons at 0x7ff53dcfd4c0> for event 'html-page-context' threw an exception (exception: 'sourcename')
make: *** [Makefile:23: html] Fehler 2
```

The traceback seems to show that the error comes from Sphinx Book theme. The theme expects to `sourcename` key in the context in [src/sphinx_book_theme/header_buttons/__init__.py#L152](https://github.com/executablebooks/sphinx-book-theme/blob/master/src/sphinx_book_theme/header_buttons/__init__.py#L152). But, the `sourcename` key is removed from the context for uncommitted files in https://github.com/mgeier/sphinx-last-updated-by-git/blob/master/src/sphinx_last_updated_by_git.py#L222.

What is the purpose of removing the `sourcename` key and thus breaking other extensions that rely on it? Are there other alternatives to get the same outcome for *sphinx-last-updated-by-git* extension without breaking other extensions?

I replaced the line with `context['sourcename'] = None`. The build works and the resulting page looks fine to me.

A workaround in configuration is `git_untracked_show_sourcelink = True`.

The behavior is quite annoying during writing and with `sphinx-autobuild` activated. Of course, the problem resolves once the file is added to the Git repository. It took me quite some time to hunt down the root cause. Maybe, this helps somebody else, even if this issue is closed without a resulting fix in the source code.

My environment:

* Python 3.9.2 on Debian Bullseye
* Sphinx: 4.5.0
* sphinx-last-updated-by-git: 0.3.2
* sphinx-book-theme: 0.3.2

--- a/src/sphinx_last_updated_by_git.py
+++ b/src/sphinx_last_updated_by_git.py
@@ -220,6 +220,7 @@ def _html_page_context(app, pagename, templatename, context, doctree):
         timestamp, show_sourcelink = data
     if not show_sourcelink:
         del context['sourcename']
+        del context['page_source_suffix']
     if timestamp is None:
         return
 

